File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.kt
class Solution {
    fun solution(t: String, p: String): Int {
        val sb = StringBuilder()
        var count = 0

        for(i in 0..t.length-p.length) {
            for(j in p.indices) {
                if(t[i+j] > p[j]) {
                    break
                }else if(t[i+j] < p[j]) {
                    count++
                    break
                }
                if(j==p.length-1 && t[j+j] <= p[j]) count++
            }
        }
        return count
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.swift
func solution(_ t:String, _ p:String) -> Int {
    let tArray = t.map { String($0) }
    var tSubstrings = Array<String>()
    
    for i in 0...t.count-p.count {
        var tSubstring = ""
        
        for j in i...i+p.count-1 {
            tSubstring += tArray[j]
        }
        
        tSubstrings.append(tSubstring)
    }
    
    return tSubstrings.filter { Int($0)! <= Int(p)! }.count
}